File: binance4j-vision/src/main/java/com/binance4j/vision/param/CandlestickParams.java
package com.binance4j.vision.param;

import java.util.List;

import com.binance4j.core.dto.Candle;
import com.binance4j.core.exception.ApiException;

import okhttp3.ResponseBody;
import retrofit2.Call;

/** The candlestick request executor */
public class CandlestickParams extends VisionParams<Candle> {
	/**
	 * @param call The API call.
	 */
	public CandlestickParams(Call<ResponseBody> call) {
		super(call);
	}

	protected List<Candle> csvToObject(List<List<String>> input) throws ApiException {
		return csvToObject(Candle.class, input);
	}
}

File: src/main/kotlin/com/binance4j/connectors/vision/param/CandlestickParams.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.connectors.vision.param

import com.binance4j.connectors.core.dto.Candle
import com.binance4j.connectors.core.exception.ApiException
import okhttp3.ResponseBody
import retrofit2.Call

/** The candlestick request executor.
 * @param call API call.
 */
class CandlestickParams(call: Call<ResponseBody>) : VisionParams<Candle>(call) {
	@Throws(ApiException::class)
	override fun csvToObject(input: List<List<String>>): List<Candle> = csvToObject(Candle::class.java, input)
}